src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEMethodSymbol.cs

```diff
@@ -1239,6 +1250,20 @@ internal override bool IsDeclaredReadOnly
             }
         }
 
+        internal override bool IsInitOnly
+        {
+            get
+            {
+                if (!_packedFlags.IsInitOnlyPopulated)
+                {
+                    bool isInitOnly = this.MethodKind == MethodKind.PropertySet &&
```

Comment:
>this.MethodKind == MethodKind.PropertySet [](start = 38, length = 41)

Are we testing significance of this MethodKind check?

Reply:
Added `ModReqOnGetAccessorReturnValue` test explicitly pulling on this API

---
In reply to: [413384135](https://github.com/dotnet/roslyn/pull/43275#discussion_r413384135) [](ancestors = 413384135)